src/App.js

```diff
@@ -16,21 +16,40 @@ import {
 import './App.css';
 import cats from "./mockCats.js"
 
-function App() {
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      cats: cats
+    }
+  }
+
+
+  render () {
   return (
     <Router>
       <Header />
       <Switch>
         <Route exact path="/" component={ Home } />
-        <Route path="/catindex" component={ CatIndex } />
-        <Route path="/catshow" component={ CatShow } />
+        {/*         */}
+        <Route path="/catindex" 
+        render = { () => <CatIndex cats = {this.state.cats} />} 
+        />
+        {/*         */}
+        <Route path="/catshow/:id" render = {(props) => { const id = +props.match.params.id;
+        const foundKitty = this.state.cats.find(cat => cat.id === id);
+        return <CatShow cat = { foundKitty } /> }} />
```

Comment:
```suggestion
        <Route path="/catshow/:id" render = {(props) => { 
            const id = +props.match.params.id;
            const foundKitty = this.state.cats.find(cat => cat.id === id);
            return <CatShow cat = { foundKitty } /> }} 
        />
```

just some suggested formatting